src/tz390.java

```diff
@@ -3348,11 +3361,85 @@ public void init_tz390(){
+                }                                                                                        // #503
+            }                                                                                            // #503
+        catch (Exception e)                                                                              // #503
+           {abort_error(41,"Error in optable option definition " + optable_optable_equivalence[index] + " - " + e.toString()); // #503
+            }                                                                                            // #503
+        index++;                                                                                         // #503
+        }                                                                                                // #503
+    index=0;                                                                                             // #503
+    while (index < machine_optable_equivalence.length) // for each machine option                        // #503
+       {try      // separate entry into machine id part and associated optable value                     // #503
+           {entry=machine_optable_equivalence[index].toUpperCase();                                      // #503
+            i=entry.indexOf("=");                                                                        // #503
+            if (i == -1)                                                                                 // #503
+               {abort_error(40,"Missing equal sign in machine option definition " + entry);              // #503
+                }                                                                                        // #503
+            machine_option_id[index]=entry.substring(0,i);                                               // #503
+            machines_optable [index]=entry.substring(i+1);                                               // #503
+            // issue error when the optable is not defined                                               // #503
+            index2=0;                                                                                    // #503
+            while (index2 < optables_optable.length) // check all optable entries defined                // #503
+               {if (machines_optable[index].equals(optables_optable[index2]))                            // #503
+                   {break;                                                                               // #503
+                    }                                                                                    // #503
+                index2++;                                                                                // #503
+                }                                                                                        // #503
+            if (index2 >= optables_optable.length) // optable is defined?                                // #503
```

Comment:
"optable is defined?" should be ??? "optable is not defined?"

Reply:
I have changed the comment to: invalid index indicates not-found condition

Reply:
Comment not changed??? My interpretation of the "//comment" on the "if ..." line

if ( condition )  // comment
{
    code if condition is true
}

explains what condition==true means. In your case, "index2 >= optables_optable.length" is checking to see if the optable is not defined -- no match in the while loop. 

I guess we disagtee on what the comment should say.

Reply:
Changed comments have not yet been pushed. Waiting for Anthony to complete our discussion on the git and gradle commands quickref. I'll push the changes once all is done. Thanks for your approval.